### PR TITLE
PC: azure_cli: Repair pint command

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [bsc#1201870c1](https://bugzilla.suse.com/show_bug.cgi?id=1201870#c1)
- Issue: [OSD](https://openqa.suse.de/tests/9121737#step/azure_cli/88)
- Verification run: [pdostal-server](https://pdostal-server.suse.cz/tests/241)

### CODE REVIEW

File: tests/publiccloud/azure_cli.pm [hunk 1]
```diff
@@ -46,7 +46,7 @@ sub run {
     assert_script_run("az group create -n $resource_group --tags '$tags'");
 
     # Pint - command line tool to query pint.suse.com to get the current image name
-    my $image_name = script_output("pint microsoft images --active --json | jq -r '.images[] | select( (.urn | contains(\"sles-15-sp3:gen2\")) and (.state == \"active\") and (.environment == \"PublicAzure\")).urn'");
+    my $image_name = script_output("pint microsoft images --active --json | jq -r '.images[] | select( (.urn | contains(\"sles-15-sp4\")) and (.state == \"active\")).urn' | head -n1");
```
Comment (jlausuch) [inline tests/publiccloud/azure_cli.pm:49]: Can you add a check that `$image_name` is not empty?

Comment (pdostal) [reply tests/publiccloud/azure_cli.pm:49]: Done.

File: tests/publiccloud/azure_cli.pm [hunk 2]
```diff
@@ -46,7 +46,8 @@ sub run {
     assert_script_run("az group create -n $resource_group --tags '$tags'");
 
     # Pint - command line tool to query pint.suse.com to get the current image name
-    my $image_name = script_output("pint microsoft images --active --json | jq -r '.images[] | select( (.urn | contains(\"sles-15-sp3:gen2\")) and (.state == \"active\") and (.environment == \"PublicAzure\")).urn'");
+    my $image_name = script_output(qq/pint microsoft images --active --json | jq -r '[.images[] | select( .urn | contains("sles-15-sp4:gen2") )][0].urn'/);
+    die("The pint query output is empty.") if ($image_name);
```
Comment (jlausuch) [inline tests/publiccloud/azure_cli.pm:50]: `unless`?

Comment (pdostal) [reply tests/publiccloud/azure_cli.pm]: Oh :/ Fixed now.

**Review by jlausuch — APPROVED** [review]

**Comment by pdostal** [comment]
@jlausuch there is unrelated issue, look the VR:
```bash
# az group create -n openqa-cli-test-rg-191 --tags 'openqa-cli-test-tag=191 openqa_created_by=openqa-vm openqa_ttl=7500'; echo LZffI-$?-
CLIInternalError: The command failed with an unexpected error. Here is the traceback:
invalid syntax (custom.py, line 839)
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/knack/cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/lib/python3.4/site-packages/azure/cli/core/commands/__init__.py", line 558, in execute
    self.commands_loader.load_arguments(command)
  File "/usr/lib/python3.4/site-packages/azure/cli/core/__init__.py", line 478, in load_arguments
    self.command_table[command].load_arguments()  # this loads the arguments via reflection
  File "/usr/lib/python3.4/site-packages/azure/cli/core/commands/__init__.py", line 315, in load_arguments
    super(AzCliCommand, self).load_arguments()
  File "/usr/lib/python3.4/site-packages/knack/commands.py", line 106, in load_arguments
    cmd_args = self.arguments_loader()
  File "/usr/lib/python3.4/site-packages/azure/cli/core/__init__.py", line 787, in default_arguments_loader
    op = handler or self.get_op_handler(operation, operation_group=kwargs.get('operation_group'))
  File "/usr/lib/python3.4/site-packages/azure/cli/core/__init__.py", line 827, in get_op_handler
    op = import_module(mod_to_import)
  File "/usr/lib64/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1467, in exec_module
  File "<frozen importlib._bootstrap>", line 1572, in get_code
  File "<frozen importlib._bootstrap>", line 1532, in source_to_code
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/usr/lib/python3.4/site-packages/azure/cli/command_modules/resource/custom.py", line 839
    err_messages = [f'{preflight_error.code} - {preflight_error.message}']
                                                                        ^
SyntaxError: invalid syntax
To open an issue, please run: 'az feedback'
LZffI-1-
```

**Comment by jlausuch** [comment]
> @jlausuch there is unrelated issue, look the VR:
> 
> ```shell
> # az group create -n openqa-cli-test-rg-191 --tags 'openqa-cli-test-tag=191 openqa_created_by=openqa-vm openqa_ttl=7500'; echo LZffI-$?-
> CLIInternalError: The command failed with an unexpected error. Here is the traceback:
> invalid syntax (custom.py, line 839)
> Traceback (most recent call last):
>   File "/usr/lib/python3.4/site-packages/knack/cli.py", line 215, in invoke
>     cmd_result = self.invocation.execute(args)
>   File "/usr/lib/python3.4/site-packages/azure/cli/core/commands/__init__.py", line 558, in execute
>     self.commands_loader.load_arguments(command)
>   File "/usr/lib/python3.4/site-packages/azure/cli/core/__init__.py", line 478, in load_arguments
>     self.command_table[command].load_arguments()  # this loads the arguments via reflection
>   File "/usr/lib/python3.4/site-packages/azure/cli/core/commands/__init__.py", line 315, in load_arguments
>     super(AzCliCommand, self).load_arguments()
>   File "/usr/lib/python3.4/site-packages/knack/commands.py", line 106, in load_arguments
>     cmd_args = self.arguments_loader()
>   File "/usr/lib/python3.4/site-packages/azure/cli/core/__init__.py", line 787, in default_arguments_loader
>     op = handler or self.get_op_handler(operation, operation_group=kwargs.get('operation_group'))
>   File "/usr/lib/python3.4/site-packages/azure/cli/core/__init__.py", line 827, in get_op_handler
>     op = import_module(mod_to_import)
>   File "/usr/lib64/python3.4/importlib/__init__.py", line 109, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
>   File "<frozen importlib._bootstrap>", line 1129, in _exec
>   File "<frozen importlib._bootstrap>", line 1467, in exec_module
>   File "<frozen importlib._bootstrap>", line 1572, in get_code
>   File "<frozen importlib._bootstrap>", line 1532, in source_to_code
>   File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
>   File "/usr/lib/python3.4/site-packages/azure/cli/command_modules/resource/custom.py", line 839
>     err_messages = [f'{preflight_error.code} - {preflight_error.message}']
>                                                                         ^
> SyntaxError: invalid syntax
> To open an issue, please run: 'az feedback'
> LZffI-1-
> ```

is it because the command `az group create -n openqa-cli-test-rg-191 --tags 'openqa-cli-test-tag=191 openqa_created_by=openqa-vm openqa_ttl=7500'` is wrong?

**Comment by pdostal** [comment]
> is it because the command `az group create -n openqa-cli-test-rg-191 --tags 'openqa-cli-test-tag=191 openqa_created_by=openqa-vm openqa_ttl=7500'` is wrong?

I don't think so. I can run this command from my Leap 15.4 machine successfully.

**Comment by ilausuch** [comment]
Maybe the package version of python3-azure-mgmt-resource is not compatible with the current azure-cli?

**Pull request merged**